django_filters/fields.py

```diff
@@ -257,16 +269,21 @@ def __init__(self, *args, **kwargs):
 
         super().__init__(*args, **kwargs)
 
-    def _get_choices(self):
-        return super()._get_choices()
-
-    def _set_choices(self, value):
-        super()._set_choices(value)
-        value = self.iterator(self, self._choices)
-
-        self._choices = self.widget.choices = value
-
-    choices = property(_get_choices, _set_choices)
+    @property
+    def choices(self):
+        return super().choices
```

Comment:
By redefining the property here - even though it's effectively a no-op - we can use `@choices.setter` without worrying about where it's from compared to #1606 where we had to use `@forms.ChoiceField.choices.setter` or `@forms.ModelChoiceField.choices.setter` and have identical copy-pasted setters in each subclass. Note that `super().choices` works fine, even in the mixin and subclasses will resolve to the correct getter.

Reply:
Good 👍 I knew it was worth asking your take 😜